docs/rules/periphery/p021-nwell_dotdash.csv

```diff
@@ -9,4 +9,5 @@ Nwells can merge over deep nwell if spacing too small (as in rule nwell.2)",TC,0
 (nwell.5b),nwell inside UHVI must not be on the same net as nwell outside UHVI,,N/A
 (nwell.6),Min enclosure of nwell hole by deep nwell outside UHVI,TC,1.030
 (nwell.7),"Min spacing between nwell and deep nwell on separate nets
-Spacing between nwell and deep nwell on the same net is set by the sum of the rules nwell.2 and nwell.5. By default, DRC run on a cell checks for the separate-net spacing, when nwell and deep nwell nets are separate within the cell hierarchy and are joined in the upper hierarchy. To allow net names to be joined and make the same-net rule applicable in this case, the ""joinNets"" switch should be turned on.",TC,4.500
+Spacing between nwell and deep nwell on the same net is set by the sum of the rules nwell.2 and nwell.5. By default, DRC run on a cell checks for the separate-net spacing, when nwell and deep nwell nets are separate within the cell hierarchy and are joined in the upper hierarchy. To allow net names to be joined and make the same-net rule applicable in this case, the ""joinNets"" switch should be turned on.
+waffle_chip",TC,4.500

```

Comment:
This last line here is a bit weird. I wonder what is causing it? Let's log a bug to fix it?

Reply:
Right on - my first impression is that it's a consequence of the `utf8` change -- I'd imagine that it's just an environment difference between whatever encoding your CICD is using versus my Windows machine here.  

So, I figure either I can try removing the explicit `utf8` references to see if that fixes it in your automation, or we can try to make it work with the explicit `utf8` encoding.

Reply:
As we always want the rst files to be in unix style we might be required to not use `os.path.join` in the output pathway?

Reply:
On windows Python, does the `os.posixpath` module exist?

Reply:
Looked into this one a bit more.

I think it's just an irregularity with how the CSV is formatted:

![image](https://user-images.githubusercontent.com/855559/94367245-16f8b780-00a3-11eb-97d0-aaea720b5a99.png)

Is that second line with the quotes intended to be Rule `(nwell.8)`? If so, then we can probably fix this by just adding an 8 to the first column of that row.  

If you want us to ignore this one, then I can add a check to filter it out (though I don't see what the logic for that in general would be, so it'd likely come up again in the future from a maintenance perspective).

Reply:
We have the same thing happen with rule `(nwell.5)` on L125, where it has a second row of the spreadsheet that append sit onto the previous one.  So if this behavior is intended, maybe all we need to do is just strip off the `\n`s to make it one line and not actually just filter it out.

Reply:
Those `\n`s are being added here, for reference:

![image](https://user-images.githubusercontent.com/855559/94367566-24af3c80-00a5-11eb-8a9a-08f187884e30.png)

So, probably the solution is just to add another check so that we don't allow this to happen for this particular case.

Reply:
@mithro I need some guidance for what to do in the situation described above -- is this "waffle_chip" meant to be included as part of the rule? Or is it meant to be ignored, since it seems to be in its own isolated section of the grid, a couple of rows down?

Reply:
For demonstration:

<img width="1154" alt="image" src="https://user-images.githubusercontent.com/855559/96038368-a389d080-0e2c-11eb-99d9-8821e98c8849.png">

This logic is the reason for these line breaks in the rules, like so:

<img width="639" alt="image" src="https://user-images.githubusercontent.com/855559/96038466-c916da00-0e2c-11eb-86b9-35517ee2818c.png">